### DISCUSSION TITLE
Pre configured ID\Relay Server in client

### DISCUSSION THREAD

**Discussion by DylanHill25** [rustdesk/rustdesk#4403] [General]
Hi there how you doing guys?

I was thinking if its possible to make something like this
I'm going to self host RustDesk for me and my clients but setting up ID\Relay Server may be difficult to clients or time consuming is there any way to pre configure it so when my client downloads RustDesk client he should not change anything its will be ready to make connection if that is possible how should i do that does anyone have any idea?

**Comment by SergeyMy** [comment]
**start-client-port.sh**
```
#!/usr/bin/env bash
#
#
[ -f ~/.config/rustdesk/RustDesk2.toml ] && exit
mkdir ~/.config/rustdesk
cp -af RustDesk2.toml ~/.config/rustdesk/RustDesk2.toml
./rustdesk

```
**RustDesk2.toml**
```
rendezvous_server = 'your-server:21116'

[options]
key = 'your public key'
relay-server = 'your-server'
custom-rendezvous-server = 'your-server'

```

**Comment by DylanHill25** [comment]
Thanks for your answer Sergey can you tell me how to do it exactly?

**Reply by SergeyMy** [reply]
> Thanks for your answer Sergey can you tell me how to do it exactly?
for Linux:
1. create 2 files (above)
2. unpack rustdesk here
3. run start-client-port.sh

PS for windows, everything is the same, only the start-client-port.sh file should be slightly different (in terms of commands and configuration locations)

**Comment by YamatoRyou** [comment]
My solution to this problem on Windows is to virtualize the RustDesk client with some tools to quickly deliver the client with a custom configuration to the target customer.
I think it is unreliable to carry the configuration information through the file name, because the limitation of Windows on the file name conflicts with some characters in the configuration information.

---
Doing this may cause the RustDesk client to be misrepresented by the anti-virus software.

**Comment by SergeyMy** [comment]
Both Windows and Linux use configuration files in the user profile. it is enough to transfer them along with the installation of rustdesk
PS they should contain only the required fields and only those that are not encrypted
